website/client/components/challenges/challengeDetail.vue

```diff
@@ -225,15 +233,69 @@ export default {
       return this.user._id === this.challenge.leader._id;
     },
   },
-  async mounted () {
-    this.challenge = await this.$store.dispatch('challenges:getChallenge', {challengeId: this.challengeId});
-    this.members = await this.$store.dispatch('members:getChallengeMembers', {challengeId: this.challengeId});
-    let tasks = await this.$store.dispatch('tasks:getChallengeTasks', {challengeId: this.challengeId});
-    tasks.forEach((task) => {
-      this.tasksByType[task.type].push(task);
-    });
+  mounted () {
+    if (!this.searchId) this.searchId = this.challengeId;
+    if (!this.challenge._id) this.loadChallenge();
+  },
+  async beforeRouteUpdate (to, from, next) {
+    this.searchId = to.params.challengeId;
+    await this.loadChallenge();
+
+    if (this.$store.state.challengeOptions.cloning) {
+      this.cloneTasks(this.$store.state.challengeOptions.tasksToClone);
+    }
+    next();
   },
   methods: {
+    cleanUpTask (task) {
+      let cleansedTask = omit(task, TASK_KEYS_TO_REMOVE);
+
+      // Copy checklists but reset to uncomplete and assign new id
+      if (!cleansedTask.checklist) cleansedTask.checklist = [];
+      cleansedTask.checklist.forEach((item) => {
+        item.completed = false;
+        item.id = uuid();
+      });
+
+      if (cleansedTask.type !== 'reward') {
+        delete cleansedTask.value;
+      }
+
+      return cleansedTask;
+    },
+    cloneTasks (tasksToClone) {
+      let clonedTasks = [];
+
+      for (let key in tasksToClone) {
+        let tasksSection = tasksToClone[key];
+        tasksSection.forEach(task => {
+          let clonedTask = cloneDeep(task);
+          clonedTask = this.cleanUpTask(clonedTask);
+          clonedTask = taskDefaults(clonedTask);
+          this.tasksByType[task.type].push(clonedTask);
+          clonedTasks.push(clonedTask);
+        });
+      }
+
+      this.$store.dispatch('tasks:createChallengeTasks', {
+        challengeId: this.searchId,
+        tasks: clonedTasks,
+      });
+    },
+    async loadChallenge () {
+      this.challenge = await this.$store.dispatch('challenges:getChallenge', {challengeId: this.searchId});
```

Comment:
probably we should make these 3 calls in parallel with promise.all

Reply:
Could catch! I'll update it